src/js/components/datepicker/index.js

```diff
@@ -94,6 +102,11 @@ export default Component.extend({
     const state = this.getState();
     state.setCurrentMonth(state.getPrevMonth());
   },
+  onSelectMonth() {
+    const state = this.getState();
+    state.setSelectedMonth(state.getCurrentMonth());
+    this.selectDate(null);
```

Comment:
this line doesn't seem necessary.

However `this.selectDate` should now call a function on the state model that both sets the date and clears the month.